packages/lodestar/src/network/util.ts

```diff
@@ -61,8 +61,9 @@ export function isRequestSingleChunk(method: Method): boolean {
   return Methods[method].responseType === MethodResponseType.SingleResponse;
 }
 
-export function eth2ResponseTimer<T>(): (source: AsyncIterable<T>) => AsyncGenerator<T> {
+export function eth2ResponseTimer<T>(stream: Stream): (source: AsyncIterable<T>) => AsyncGenerator<T> {
```

Comment:
why not receive stream abort controller (call it streamCloseController)?

I don't think we need another abort source in this method now?

Reply:
the other `abortDuplex` has `{returnOnAbort: true}`, this `abortSource` does not have that, we want pipe to throw Error in case of timeout

Reply:
> why not receive stream abort controller (call it streamCloseController)?

I'm not so sure we can use the same AbortController here. One is used to abort the original stream nicely, one is used to abort the stream with an Error.

Reply:
well if you use original one, it would close stream nicely (basically same thing as you did here on stream directly but removes dependency of this method to stream) and to a cleanup while one used here would throw for rest of methods in pipe from what I understand.

Reply:
If we use same controller, calling `controller.abort()` would both trigger abort the original iterator (which we don't want Error) and abort this internal iterator (which we want Error so we can catch in `sendRequest`) so the behavior is indeterministic.

so I'd use separate AbortControllers here unless there's performance issue later..

Reply:
Hmm it should be following flow in case of timeout:
1. streamAbortController.abort() - won't break anything, it will allow rest of transformations to end
2. timeoutController.abort() - will throw error from this function below to handle

Reply:
👍 I'll give it a try

Reply:
so this unit test failed: `can send/receive status messages from connected peers` if I use the same controller, error thrown with `response timeout`.
It's because it sent a `status` requests which is single chunk response and it calls break (now replaced by `controller.abort();continue;` and caused timeout error.